FunctionalTableData/TableSectionChangeSet.swift

```diff
@@ -134,7 +134,11 @@ public final class TableSectionChangeSet {
 				if oldSectionIndex < old.count && new[newSectionIndex].key == old[oldSectionIndex].key {
 					// Skip over equal sections
 					repeat {
-						compareRows(newRows: &newRows, oldRows: &oldRows, oldSectionIndex: oldSectionIndex, newSectionIndex: newSectionIndex)
+						if old[oldSectionIndex] != new[newSectionIndex] {
+							reloadedSections.insert(newSectionIndex)
+						} else {
+							compareRows(newRows: &newRows, oldRows: &oldRows, oldSectionIndex: oldSectionIndex, newSectionIndex: newSectionIndex)
+						}
```

Comment:
Wait? why do we need this? by changing `extension TableSection: Equatable` changes in the header will be taken into effect right? Another solution would be to no do the `extension TableSection: Equatable` but add a `compareHeader` similar to what we have here in `compareRows`.

But lastly, whatever you do in the headers needs to be done for the footers

Reply:
No, we can make the changes to `extension TableSection: Equatable`, but there is never anything that checks the section equality that will trigger a reload for that section. We need to add the section index to `reloadedSections` in order to see the changes take effect. 

A `compareHeader` would read better I think 👍
I can implement the changes for footers as well.

Reply:
But this can be simplified as you suggested, so I'll create a `compareHeaders()` or something, and just compare the states there, so that we don't need to touch the sections equatable at all.